spyder/plugins/plots/widgets/figurebrowser.py

```diff
@@ -709,13 +709,35 @@ def go_previous_thumbnail(self):
             index = self._thumbnails.index(self.current_thumbnail) - 1
             index = index if index >= 0 else len(self._thumbnails) - 1
             self.set_current_index(index)
+            self.scroll_to_item(index)
 
     def go_next_thumbnail(self):
         """Select thumbnail next to the currently selected one."""
         if self.current_thumbnail is not None:
             index = self._thumbnails.index(self.current_thumbnail) + 1
             index = 0 if index >= len(self._thumbnails) else index
             self.set_current_index(index)
+            self.scroll_to_item(index)
+
+    def scroll_to_item(self, index):
+        """Scroll to the selected item of ThumbnailScrollBar."""
+        spacing_between_items = self.scene.verticalSpacing()
+        height_view = self.scrollarea.viewport().height()
+        height_item = self.scene.itemAt(index).sizeHint().height()
+        height_view_excluding_item = height_view - height_item
```

Comment:
The following three lines could be replaced by:
`height_view_excluding_item = max(0, height_view - height_item)`